### ISSUE TITLE
Compile error on Windows with disabled stats

### ISSUE THREAD

**Issue by woodstock23** [nanomsg/nng#799]
## NNG & Platform details.
nng: 1.1.1
Windows 10
Visual Studio 2017

## Expected Behavior
nng compiles without errors

## Actual Behavior
nng fails to compile with error:
`d:\projects\nng-1.1.1\src\core\stats.h(53): error C2016: C requires that a struct or union has at least one member`

## Steps to Reproduce
Compile nng 1.1.1 on Windows with Visual Studio 2017 with disabled stats (NNG_ENABLE_STATS=OFF)

## Solution
In src/core/stats.h, change line 26 to:
```
#ifdef NNG_ENABLE_STATS
typedef struct nni_stat_item nni_stat_item;
#else
typedef char nni_stat_item;
#endif
```
switch lines 37 and 38:
```
#ifdef NNG_ENABLE_STATS
struct nni_stat_item {
```
and switch lines 52 and 53:
```
};
#endif
```

**Comment by gdamore** [comment]
Whoops.  Pretty sure that isn't the right fix, but I will look.  The problem is almost certainly not specific to Windows.

**Comment by gdamore** [comment]
Well, now looking at closer, maybe it is.  It comes up on Windows because apparently clang on mac is happy with that empty struct.

**Comment by woodstock23** [comment]
Yep, C99 doesn't allow empty structs, but many compiler don't complain or give only a warning.

**Comment by gdamore** [comment]
Yeah, I did some research on that.  I have a fix coming for this, which is similar to, but different from, the solution you proposed.  I just inject a single char member in the struct.  This way padding is performed (with zero size I wasn't worried about that, but with non-zero size I want to make sure any padding is done properly in case someone writes code to depend on it.)

I also found a few other compiler complaints (unused variables) in pipe.c which I'm going to fix at the same time.
